packages/client/lib/miner/miner.ts

```diff
@@ -0,0 +1,258 @@
+      // Determine if signer is INTURN (2) or NOTURN (1)
+      const inTurn = await vmCopy.blockchain.cliqueSignerInTurn(signerAddress)
+      difficulty = inTurn ? 2 : 1
+    }
+
+    let baseFeePerGas
+    const londonHardforkBlock = this.config.chainCommon.hardforkBlockBN('london')
+    const isInitialEIP1559Block = londonHardforkBlock && number.eq(londonHardforkBlock)
+    if (isInitialEIP1559Block) {
+      baseFeePerGas = new BN(this.config.chainCommon.param('gasConfig', 'initialBaseFee'))
+    } else if (this.config.chainCommon.isActivatedEIP(1559)) {
+      baseFeePerGas = parentBlockHeader.calcNextBaseFee()
+    }
+
+    const parentBlock = (this.synchronizer as any).chain.blocks.latest
+    const blockBuilder = await vmCopy.buildBlock({
+      parentBlock,
+      headerData: {
+        number,
+        difficulty,
+        gasLimit,
+        baseFeePerGas,
+      },
+      blockOpts: {
+        cliqueSigner: signerPrivKey,
+        hardforkByBlockNumber: true,
+      },
+    })
+
+    const txs = await this.synchronizer.txPool.txsByPriceAndNonce(
+      vmCopy.stateManager,
+      baseFeePerGas
+    )
+    this.config.logger.info(
+      `Miner: Assembling block from ${txs.length} eligible txs ${
+        baseFeePerGas ? `(baseFee: ${baseFeePerGas.toNumber()})` : ''
+      }`
+    )
+    let index = 0
+    let blockFull = false
+    while (index < txs.length && !blockFull && !interrupt) {
+      try {
+        await blockBuilder.addTransaction(txs[index])
+      } catch (error: any) {
+        if (error.message === 'tx has a higher gas limit than the remaining gas in the block') {
+          if (blockBuilder.gasUsed.gt(gasLimit.subn(21000))) {
+            // If block has less than 21000 gas remaining, consider it full
+            blockFull = true
+            this.config.logger.debug(
+              `Miner: Assembled block full (gasLeft: ${gasLimit
+                .sub(blockBuilder.gasUsed)
+                .toNumber()})`
+            )
+          }
+        } else {
+          // If there is an error adding a tx, it will be skipped
+          const hash = '0x' + txs[index].hash().toString('hex')
+          this.config.logger.error(
+            `Skipping tx ${hash}, error encountered when trying to add tx:\n${error}`
+          )
+        }
+      }
+      index++
+    }
+    if (interrupt) return
+    // Build block, sealing it
+    const block = await blockBuilder.build()
+    this.config.logger.info(`Miner: Sealed block with ${block.transactions.length} txs`)
+    this.assembling = false
+    if (interrupt) return
+    // Put block in blockchain
+    await this.synchronizer.handleNewBlock(block)
+    // Remove included txs from TxPool
+    this.synchronizer.txPool.removeNewBlockTxs([block])
+    // Inform connected peers of new block
+    for (const peer of (this.synchronizer as any).pool.peers) {
+      const { td } = (this.synchronizer as any).chain.headers
+      peer.eth.send('NewBlock', [block, td])
+    }
```

Comment:
This peer informing is also done in `synchronizer.handleNewBlock()`, so it is redundant if I am not overlooking or misreading something.

Reply:
yes thanks, will remove